### ISSUE TITLE
[REVIEW]: swsc: A sitewise UCE partitioner

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1146]
**Submitting author:** @RHagenson (<a href="http://orcid.org/0000-0001-9750-1925">Ryan Hagenson</a>)
**Repository:** <a href="https://bitbucket.org/RHagenson/swsc/" target ="_blank">https://bitbucket.org/RHagenson/swsc/</a>
**Version:** v6.0.1
**Editor:** @pjotrp
**Reviewer:** @urbanslug
**Archive:** <a href="https://doi.org/10.5281/zenodo.2542907" target="_blank">10.5281/zenodo.2542907</a>

## Status

[![status](http://joss.theoj.org/papers/b01aa9474605d9516d710ae25b6b8168/status.svg)](http://joss.theoj.org/papers/b01aa9474605d9516d710ae25b6b8168)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/b01aa9474605d9516d710ae25b6b8168"><img src="http://joss.theoj.org/papers/b01aa9474605d9516d710ae25b6b8168/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/b01aa9474605d9516d710ae25b6b8168/status.svg)](http://joss.theoj.org/papers/b01aa9474605d9516d710ae25b6b8168)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://bitbucket.org/RHagenson/swsc/" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@urbanslug, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.theoj.org/about#reviewer_guidelines. Any questions/concerns please let @pjotrp know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://bitbucket.org/RHagenson/swsc/">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** Does the release version given match the GitHub release (v6.0.1)?
- [x] **Authorship:** Has the submitting author (@RHagenson) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @urbanslug it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01146/joss.01146/10.21105.joss.01146.pdf)

**Comment by pjotrp** [comment]
@RHagenson we are starting review in this issue tracker. To expedite the review process do you mind going through above list of check boxes and make sure they can be ticked (you can't tick them). Also check the PDF output carefully. Ping us here when you are done.

**Comment by RHagenson** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01146/joss.01146/10.21105.joss.01146.pdf)

**Comment by RHagenson** [comment]
@pjotrp and @urbanslug, I have reviewed the checklist above, made corrections to the PDF, and deleted the tags containing a major bug previously discussed in #1116 The version under review should be v6.1.0, which contains the bug fix that caused overlapping UCE partitions.

**Comment by pjotrp** [comment]
Thank you @RHagenson. Before we start review, can you explain the license? It looks like MIT/expat. If so, I think you should make it explicit.

**Comment by RHagenson** [comment]
@pjotrp and @urbanslug I updated the README to clarify the license as Modified BSD License.

**Comment by pjotrp** [comment]
Excellent. @urbanslug you can start review!

**Comment by RHagenson** [comment]
@urbanslug and @pjotrp, the performance claims are bold, however they are entirely algorithmic. 

Going from a brute-force "consider every window of at least size `--minWin`" to the candidate window plus extension procedure had a severe reduction in the amount of work needed to get approximate results. Reading the code should be sufficient proof, but I also offer that my real dataset had 39966 positions across 50 UCEs (~799 avg length) where I selected minWin to be 100 and number of candidates to be 5. 

I will check with the biologist on this project about sharing the real dataset for validation purposes as it is not my data to share.

**Comment by pjotrp** [comment]
Ping @RHagenson - final step takes just a few minutes!

**Comment by RHagenson** [comment]
@pjotrp I apologize about the delay, been terribly ill all week so just today got caught up on urgent affairs enough to tackle these final steps. The Zenedo DOI is: 10.5281/zenodo.2542907

**Comment by pjotrp** [comment]
@whedon set 10.5281/zenodo.2542907 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.2542907" target="_blank">10.5281/zenodo.2542907</a> is the archive.

**Comment by pjotrp** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by pjotrp** [comment]
@arfon we are ready to roll! Thanks @RHagenson and @george-githinji and @urbanslug!

**Comment by arfon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/442
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01146](https://doi.org/10.21105/joss.01146)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by arfon** [comment]
@urbanslug - many thanks for reviewing this submission and to @pjotrp and @george-githinji for editing this one ✨ 

@RHagenson - your paper is now accepted into JOSS :zap::rocket::boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](http://joss.theoj.org/papers/10.21105/joss.01146/status.svg)](https://doi.org/10.21105/joss.01146)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01146">
  <img src="http://joss.theoj.org/papers/10.21105/joss.01146/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: http://joss.theoj.org/papers/10.21105/joss.01146/status.svg
   :target: https://doi.org/10.21105/joss.01146
```
This is how it will look in your documentation:

[![DOI](http://joss.theoj.org/papers/10.21105/joss.01146/status.svg)](https://doi.org/10.21105/joss.01146)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: http://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://www.flipcause.com/secure/cause_pdetails/Mjk3Nzk=

**Comment by RHagenson** [comment]
@pjotrp Is there a policy for submitting corrections/edits after acceptance? Of course, the corrections/edits do not change the usage of the tool, that is still as it has been. Rather, I want to change the official repo from BitBucket to GitHub and I realized I misspelled a name in my acknowledgements (Sefszek -> Sefczek).

**Comment by arfon** [comment]
@RHagenson - we can fix a typo for sure, and update the repository too. Can you make the update to the `paper.md` file and let me know when you've done this?

**Comment by RHagenson** [comment]
@pjotrp I have updated the `paper.md` with the corrections and updated all the files to refer to `github.com/rhagenson/swsc` rather than `bitbucket.org/rhagenson/swsc` I also pushed a new `v6.1.2` tag to begin a new release of the code. Anything else you need from me?

**Comment by arfon** [comment]
@RHagenson - the paper is now updated with these changes. The version showing on the JOSS site can take a few hours to update as there is caching in place.

**Comment by RHagenson** [comment]
@arfon Not to be a bother, but I was looking at the rendered [paper](https://joss.theoj.org/papers/10.21105/joss.01146) on JOSS the other day and noticed in the first paragraph most of the citations did not get formatted properly. Is this a known issue? What do you need me to do to get this fixed?

**Comment by arfon** [comment]
Oh yeah, that’s weird. I’ll investigate and report back.

**Comment by arfon** [comment]
> @arfon Not to be a bother, but I was looking at the rendered paper on JOSS the other day and noticed in the first paragraph most of the citations did not get formatted properly. Is this a known issue? What do you need me to do to get this fixed?

Can you check again now? The PDF looks better now to me: https://github.com/openjournals/joss-papers/blob/ca9a5de11383d56589212a7c67079fa2ce883c0b/joss.01146/10.21105.joss.01146.pdf

The PDF on the live site might take a few hours to update as there's caching in place.

**Comment by RHagenson** [comment]
@arfon That looks right now! I will check the PDF on the live site some time later to ensure it updated there. Thank you for introducing the fix!
